federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/csw/persist/CreateCswSourcePersistMethod.java

```diff
@@ -97,4 +106,15 @@ public Report persist(CswSourceConfiguration configuration) {
         }
         return validationResults;
     }
+
+    @Override
+    public List<ConfigurationMessage> validateRequiredFields(
+            CswSourceConfiguration configuration) {
+        Report report = handler.test(SOURCE_NAME_EXISTS_TEST_ID, configuration);
```

Comment:
The messages from this test and from calling super.validateRequiredFields should be returned in one list. This allows the user to fix all invalid errors at once.